### ISSUE TITLE
Spawning an effect with Visibility::Hidden causes Hanabi to panic

### ISSUE THREAD

**Issue by wilk10** [djeedai/bevy_hanabi#182]
Thanks a lot for working on Hanabi. I have run into the following issue, thanks for looking into it:

**Crate versions**
```toml
bevy = "0.10.1"
# hanabi main
bevy_hanabi = { git = "https://github.com/djeedai/bevy_hanabi", rev = "e0ebf1b147e20241e2ffad21528a58d9df2ea376" }
```

**Describe the bug**
I would like to be able to spawn an effect that has initially `Visibility::Hidden`, and later make it visible (my use-case is that it depends on the visibility of the parent sprite, but the bug applies also to effects without parent).

What happens instead is that when the effect is spawned, Hanabi panics with the following error:
```
thread 'Compute Task Pool (1)' panicked at 'called `Option::unwrap()` on a `None` value', C:\Users\anssa\.cargo\git\checkouts\bevy_hanabi-859bacb1c5d664ab\e0ebf1b\src\render\mod.rs:1173:52
```
The culprit is the system `extract_effects`, where the query for newly `Added` effects contains the effect, but its handle is not found in the extracted effects `effects: Extract<Res<Assets<EffectAsset>>>` and panics on `let asset = effects.get(&effect.asset).unwrap();`

 If it's spawned as `Visibility::Visible` and later hidden, it's all fine, like in the `visibility.rs` example.

**Expected behavior**
I would expect the effect to spawn and have `Visibility::Hidden`, and later become visible based on game logic.

**To Reproduce**
```rust
use bevy::prelude::*;
use bevy_hanabi::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_plugin(HanabiPlugin)
        .add_startup_system(setup)
        .run();
}

fn setup(
    mut commands: Commands,
    mut effects: ResMut<Assets<EffectAsset>>,
) {
    commands.spawn(Camera2dBundle::default());

    let effect = EffectAsset {
        capacity: 4096,
        spawner: Spawner::rate(50.0.into()),
        simulation_condition: SimulationCondition::WhenVisible,
        ..Default::default()
    };

    let handle = effects.add(effect);

    commands
        .spawn(ParticleEffectBundle {
            visibility: Visibility::Hidden, // <== spawned as Hidden
            effect: ParticleEffect::new(handle),
            ..default()
        });
}
```

**Screenshots**
Not relevant.

**Comment by djeedai** [comment]
Thanks, I will have a look. Should be easy to fix.

**Comment by djeedai** [comment]
This should be fixed (and tested) with #193. Closing for now, let me know if there's still an issue. Thanks!
